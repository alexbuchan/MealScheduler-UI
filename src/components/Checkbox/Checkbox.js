import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  label: PropTypes.string,  // The name on the label that renders in the UI
  name: PropTypes.string,   // The name of the field
  type: PropTypes.string,   // The type of input for the field
  value: PropTypes.string,  // The value of the input in the field
  onChange: PropTypes.func,  // The onChange function for updating the field
  validationField: PropTypes.object   // Validation object returned from the Validator instance.
};

const Checkbox = ({ label, name, type, value, onChange, validationField }) => {
  const withValidation = () => {
    return (
      <div className="form-element">
        <label>{ label }</label>
        <input name={ name } type={ type } defaultChecked={ value } onChange={ onChange } />
      </div>
    );
  }

  const noValidation = () => {
    return (
      <div className="form-element">
        <label>{ label }</label>
        <input name={ name } type={ type } defaultChecked={ value } onChange={ onChange } />
      </div>
    );
  }

  if (validationField) {
    return withValidation();
  }

  return noValidation();
}

Checkbox.propTypes = propTypes;
export default Checkbox;