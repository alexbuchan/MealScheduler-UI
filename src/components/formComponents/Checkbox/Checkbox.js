import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  label: PropTypes.string,              // The name on the label that renders in the UI. /Default empty string
  name: PropTypes.string.isRequired,    // The name of the field.
  type: PropTypes.string.isRequired,    // The type of input for the field.
  value: PropTypes.bool,                // The value of the input in the field. /Default is false
  onChange: PropTypes.func.isRequired   // The onChange function for updating the field.
};

const Checkbox = ({ label, name, type, value, onChange }) => {
  return (
    <div className="form-element">
      <label>{ label }</label>
      <input name={ name } type={ type } defaultChecked={ value } onChange={ onChange } />
    </div>
  );;
}

Checkbox.defaultProps = {
  label: '',
  value: false
}

Checkbox.propTypes = propTypes;
export default Checkbox;