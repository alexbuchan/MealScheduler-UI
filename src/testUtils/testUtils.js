import React from 'react';
import { shallow } from 'enzyme';

export const componentSetup = (Component, props = {}, options = {}) => {
  const instance = shallow(<Component { ...props } />, options);
  return instance;
}

export const findByTestAttribute = (instance, attribute) => {
  const wrapper = instance.find(`[data-test="${attribute}"]`);
  return wrapper;
}