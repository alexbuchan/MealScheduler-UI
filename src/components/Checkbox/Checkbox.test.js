import Checkbox from './Checkbox';
import { componentSetup, findByTestAttribute } from '../../testUtils/testUtils';

describe('Checkbox', () => {
  describe('When no label is specified', () => {
    let instance;

    beforeEach(() => {
      const props = {
        name: 'name',
        type: 'text',
        onChange: () => {}
      };

      instance = componentSetup(Checkbox, props);
    });
    
    it('Should not render a label', () => {
      expect(instance).toMatchSnapshot();
    });
  });

  describe('When no value is specified', () => {
    let instance;

    beforeEach(() => {
      const props = {
        name: 'name',
        type: 'text',
        onChange: () => {}
      };

      instance = componentSetup(Checkbox, props);
    });
    
    it('Value should be false', () => {
      expect(instance).toMatchSnapshot();
    });
  });

  describe('When all props are passed', () => {
    let instance;

    beforeEach(() => {
      const props = {
        label: 'label',
        name: 'name',
        type: 'text',
        value: false,
        onChange: () => {}
      };

      instance = componentSetup(Checkbox, props);
    });
    
    it('Should render correctly', () => {
      expect(instance).toMatchSnapshot();
    });
  });
});