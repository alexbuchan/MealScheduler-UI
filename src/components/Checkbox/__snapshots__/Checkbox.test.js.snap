// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox When all props are passed Should render correctly 1`] = `
<div
  className="form-element"
>
  <label>
    label
  </label>
  <input
    defaultChecked={false}
    name="name"
    onChange={[Function]}
    type="text"
  />
</div>
`;

exports[`Checkbox When no label is specified Should not render a label 1`] = `
<div
  className="form-element"
>
  <label />
  <input
    defaultChecked={false}
    name="name"
    onChange={[Function]}
    type="text"
  />
</div>
`;

exports[`Checkbox When no value is specified Value should be false 1`] = `
<div
  className="form-element"
>
  <label />
  <input
    defaultChecked={false}
    name="name"
    onChange={[Function]}
    type="text"
  />
</div>
`;
